keps/sig-api-machinery/20180415-crds-to-ga.md

```diff
@@ -143,22 +144,29 @@ The CRD `v1` API will be the same as the `v1beta1` but with all changes to the A
 
 ### Integration tests for GA
 
-TODO: complete this list
-
-* [Pruning for CustomResources KEP](https://github.com/kubernetes/enhancements/pull/709)
-  * TODO
-* [Vanilla OpenAPI Subset Design](https://docs.google.com/document/d/1pcGlbmw-2Y0JJs9hsYnSBXamgG9TfWtHY6eh80zSTd8)
-  * TODO
-* CRD versioning/conversion (https://github.com/kubernetes/kubernetes/issues/64136):
+* Ensure CRD themselves are v1<->v1beta1 round trippable
+* CRD versioning/conversion integration test coverage:
   * Ensure what is persisted in etcd matches the storage version
   * Set up a CRD, persist some data, changed the version, and ensure the previously persisted data is readable
   * Ensure discovery docs track a CRD through creation, version addition, version removal, and deletion
   * Ensure `spec.served` is respected
+  * Ensure illegal inputs result in sensible errors, not panics, and that the system recovers appropriately
+    * CRD mutation operations: invalid schemas, misconfigured webhooks, scope changes, ...
+    * webhook conversion responses: conversion responses that do not align with request (uuid, list size and elements, ..)
+* See also [Pruning for Custom Resources KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20180731-crd-pruning.md)
+* See also [Vanilla OpenAPI Subset: structural schemas KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190425-structural-openapi.md)
+* See also [Defaulting for Custom Resources KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190426-crd-defaulting.md)
 
 ### e2e tests for GA
 
 * Custom Resources should be readable and writable at all available versions (test for round trip-ability)
-* A Custom Resource watch should terminate if its CustomResourceDefinition is deleted or updated
+* Introduce v1 API testing in addition to v1beta1 tests
```

Comment:
Will a v1beta1 CRD continue functioning in the case where the control plane version with v1 is rolled back?

Reply:
yes. no new fields are being added, and v1beta1 validation tolerates all values being permitted in v1

Reply:
unit tests cover the skew validation cases for updating via v1beta1 a CRD persisted via v1